src/library/Checkbox/Checkbox.js

```diff
@@ -127,4 +191,46 @@ export default class Checkbox extends PureComponent<Props> {
 
     return <Root {...rootProps} />;
   }
+
+  setInputRef = (node: HTMLInputElement) => {
+    const { inputRef } = this.props;
+    this.input = node;
+    inputRef && inputRef(node);
+  };
+
+  updateState = (indeterminate: boolean, checked: boolean) => {
+    this.indeterminate = indeterminate;
+    if (this.input) {
+      this.input.indeterminate = indeterminate;
+      this.input.checked = checked;
+    }
+  };
+
+  handleClick = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    const { defaultIndeterminate, indeterminate, onChange } = this.props;
+
+    if (defaultIndeterminate || indeterminate) {
+      const nextIndeterminate = this.isControlled('indeterminate')
+        ? this.getIndeterminateValue()
+        : !this.getIndeterminateValue();
+
+      this.updateState(nextIndeterminate, nextIndeterminate);
```

Comment:
On first glance, this is a little confusing to me, but not sure if it's worth an extra line to indicate that checked depends on indeterminate here.

Reply:
Yeah, I could go either way...